### ISSUE TITLE
Unexpected token in IMAP response: [atom: 0]

### ISSUE THREAD

**Issue by mike-nelson** [jstedfast/MailKit#428]
Hi, I got this error in this case from an IMAP server called Courier-IMAP (version from 2004) which is running in the wild (this is a user - not my server).

Trying to remove a flag from an email - seems to be OK but MailKit reports this as an error.

```
S: Q00000006 OK [READ-WRITE] SELECT completed.
C: Q00000007 UID STORE 16334 -FLAGS (\Flagged)
S: * 0 FETCH (UID 16334 FLAGS ())
S: Q00000007 OK STORE completed.
```

`Unexpected token in IMAP response: [atom: 0]`

I'm just going to catch that error on that line and call it OK, just thought you might like to know this case.

**Comment by jstedfast** [comment]
Looks like the Courier-IMAP thinks that indexes start at 0 when in fact they start at 1 :(

My guess is that newer versions of Courier IMAP have been fixed.

I'll commit a fix to allow 0 for the index, I guess. You will probably have to ignore an Index of 0 in the MessageFlagsChanged event, though.

**Comment by jstedfast** [comment]
When Courier-IMAP sends "* 0 FETCH ...", the Index value in the event args will be `-1`.

**Comment by mike-nelson** [comment]
Thanks

Cheers
-Mike

Mike Nelson // CEO (and Developer) // Beweb <http://www.beweb.co.nz> // 09
3077042 // 027 4403757
Listen to your email with Speaking Email <https://speaking.email> by beweb

**Comment by jstedfast** [comment]
I released MailKit 1.10.1 with this fix to nuget.org

**Comment by jstedfast** [comment]
@mike-nelson would you be willing to send me the first few lines of a log connecting to Courier IMAP? I mostly care about the greeting, but the CAPABILITIES would also be nice. Thanks!
